File: src/main/java/net/atos/client/zgw/zrc/model/zaakobjecten/ZaakobjectNummeraanduiding.java
/*
 * SPDX-FileCopyrightText: 2023 Atos, 2025 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.client.zgw.zrc.model.zaakobjecten;

import java.net.URI;

import nl.info.client.zgw.zrc.model.generated.ObjectTypeEnum;

/**
 * ZaakobjectNummeraanduiding
 */
public class ZaakobjectNummeraanduiding extends ZaakobjectMetObjectIdentificatie<ObjectOverige<ObjectNummeraanduiding>> {

    public static final String OBJECT_TYPE_OVERIGE = "nummeraanduiding";

    /**
     * Constructor for JSONB deserialization
     */
    public ZaakobjectNummeraanduiding() {
    }

    /**
     * Constructor with required attributes
     */
    public ZaakobjectNummeraanduiding(final URI zaak, final URI bagObjectUri, final ObjectNummeraanduiding nummeraanduiding) {
        super(zaak, bagObjectUri, ObjectTypeEnum.OVERIGE, new ObjectOverige<>(nummeraanduiding));
        setObjectTypeOverige(OBJECT_TYPE_OVERIGE);
    }

    @Override
    public String getWaarde() {
        return getObjectIdentificatie().overigeData.getIdentificatie();
    }
}

File: src/main/kotlin/nl/info/client/zgw/zrc/model/zaakobjecten/ZaakobjectNummeraanduiding.kt
/*
 * SPDX-FileCopyrightText: 2023, 2025, 2026 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package nl.info.client.zgw.zrc.model.zaakobjecten

import nl.info.client.zgw.zrc.model.generated.ObjectTypeEnum
import java.net.URI

class ZaakobjectNummeraanduiding : ZaakobjectMetObjectIdentificatie<ObjectOverige<ObjectNummeraanduiding>> {
    companion object {
        const val OBJECT_TYPE_OVERIGE_NUMMERAANDUIDING = "nummeraanduiding"
    }

    /**
     * Constructor for JSONB deserialization
     */
    constructor() : super()

    /**
     * Constructor with required attributes
     */
    constructor(zaak: URI, bagObjectUri: URI?, nummeraanduiding: ObjectNummeraanduiding?) : super(
        zaak,
        bagObjectUri,
        ObjectTypeEnum.OVERIGE,
        ObjectOverige(nummeraanduiding)
    ) {
        objectTypeOverige = OBJECT_TYPE_OVERIGE_NUMMERAANDUIDING
    }

    override val waarde: String?
        get() = objectIdentificatie?.overigeData?.identificatie
}
